docker/Makefile

```diff
@@ -1,21 +1,32 @@
-.PHONY: build push deploy
+.PHONY: build clean
 
-PROJECT = $(shell gcloud config get-value project)
+build: build-stmp
+
+ifeq ($(IN_HAIL_CI),1)
+.PHONY: push deploy
 
+PROJECT = $(shell gcloud config get-value project)
+BASE_LATEST = gcr.io/$(PROJECT)/base
 BASE_IMAGE = gcr.io/$(PROJECT)/base:$(shell docker images -q --no-trunc base:latest | sed -e 's,[^:]*:,,')
-SPARK_BASE_IMAGE = gcr.io/$(PROJECT)/spark-base:$(shell docker images -q --no-trunc spark-base:latest | sed -e 's,[^:]*:,,')
 
-build:
+build-stmp: Dockerfile.base
```

Comment:
It also depends on every file in the context. `$(shell find . -type f)` should do it.